integration/networking/resolvconf_test.go

```diff
@@ -0,0 +1,72 @@
+package networking
+
+import (
+	"os"
+	"testing"
+
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+
+	containertypes "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/integration/internal/network"
+
+	"github.com/docker/docker/testutil/daemon"
+	"gotest.tools/v3/skip"
+)
+
+// Regression test for https://github.com/moby/moby/issues/46968
+func TestResolvConfLocalhostIPv6(t *testing.T) {
+	// No "/etc/resolv.conf" on Windows.
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+	// In rootless mode, "/etc/resolv.conf" isn't the host's, so it can't be replaced by the test.
+	skip.If(t, testEnv.IsRootless())
+
+	ctx := setupTest(t)
+
+	// Replace the test host's resolv.conf with one that only contains a loopback address.
```

Comment:
Yikes! Things could go very wrong if an unsuspecting developer tried to run the integration suite outside of a container. If no better way is implemented, could you add a sanity check to skip this test if it's not run from inside a container?
```go
_, err := os.Stat("/.dockerenv")
skip.If(t, err == nil, "outside a container")
```

Reply:
Yes, it's grim - but, not my idea! It was already happening in a few tests ...

https://github.com/moby/moby/blob/810ef4d0b626177696a32447effdf2f562de85b4/libnetwork/libnetwork_linux_test.go#L1799-L1809

https://github.com/moby/moby/blob/810ef4d0b626177696a32447effdf2f562de85b4/libnetwork/libnetwork_linux_test.go#L1876-L1886

https://github.com/moby/moby/blob/810ef4d0b626177696a32447effdf2f562de85b4/libnetwork/libnetwork_linux_test.go#L1954-L1964

So, I'll do the env-var thing, and fix those too.

Reply:
Eek, that's really scary! Thank you!

Reply:
I deleted test "TestResolvConf" - as I described in [now-deleted comments](https://github.com/moby/moby/compare/3faac6246c4cd2095307cbb1d69d110c2d60da14..fad457ace1930573920088c69260d5aa588318b4), it wasn't clear what it was trying to test. And, whatever it was trying to test is covered elsewhere.

I tidied up the other two by turning them into a table-driven test ... they didn't need to overwrite the host's resolv.conf or use the new env-var, they could just use `libnetwork.OptionOriginResolvConfPath`.